src/dfmt/ast_info.d

```diff
@@ -200,6 +206,15 @@ final class FormatVisitor : ASTVisitor
         destructor.accept(this);
     }
 
+    override void visit (const FunctionBody functionBody)
+    {
+        if (auto bd = functionBody.specifiedFunctionBody)
+        {
+            astInformation.funBodyLocations ~= bd.blockStatement.startLocation;
```

Comment:
there could be a null dereference here if bd.blockStatement is null.

However with the current grammar this case is impossible, it could only be introduced in the future. I don't think it's very likely that this will happen to blockStatement (as shortened statement got a separate AST name), so I am fine with merging without checking for null here